addons/portal/static/src/js/portal_signature.js

```diff
@@ -1,93 +1,126 @@
 
-    var ajax = require('web.ajax');
-    var base = require('web_editor.base');
     var core = require('web.core');
-    var Widget = require("web.Widget");
-    var rpc = require("web.rpc");
-
+    var dom = require('web.dom');
+    var NameAndSignature = require('portal.name_and_signature').NameAndSignature;
     var qweb = core.qweb;
+    var Widget = require('web.Widget');
+
+    var _t = core._t;
 
+    /**
+     * This widget is a signature request form. It uses
+     * @see NameAndSignature for the input fields, adds a submit
+     * button, and handles the RPC to save the result.
+     */
     var SignatureForm = Widget.extend({
         template: 'portal.portal_signature',
+        xmlDependencies: ['/portal/static/src/xml/portal_signature.xml'],
         events: {
-            'click #o_portal_sign_clear': 'clearSign',
-            'click .o_portal_sign_submit': 'submitSign',
-            'init #o_portal_sign_accept': 'initSign',
+            'click .o_portal_sign_submit': 'async _onClickSignSubmit',
         },
 
-        init: function(parent, options) {
+        /**
+         * Overridden to allow options.
+         *
+         * @constructor
+         * @param {Widget} parent
+         * @param {Object} options
+         * @param {string} options.callUrl - make RPC to this url
+         * @param {string} [options.sendLabel='Accept & Sign'] - label of the
+         *  send button
+         * @param {Object} [options.rpcParams={}] - params for the RPC
+         * @param {Object} [options.nameAndSignatureOptions={}] - options for
+         *  @see NameAndSignature.init()
+         */
+        init: function (parent, options) {
             this._super.apply(this, arguments);
-            this.options = _.extend(options || {}, {
-                csrf_token: odoo.csrf_token,
-            });
-        },
 
-        willStart: function() {
-            return this._loadTemplates();
-        },
+            this.csrf_token = odoo.csrf_token;
 
-        start: function() {
-            this.initSign();
-        },
+            this.callUrl = options.callUrl || '';
+            this.rpcParams = options.rpcParams || {};
+            this.sendLabel = options.sendLabel || _t("Accept & Sign");
 
-        // Signature
-        initSign: function () {
-            this.$("#o_portal_signature").empty().jSignature({
-                'decor-color': '#D1D0CE',
-                'color': '#000',
-                'background-color': '#fff',
-                'height': '142px',
-                'width': '100%', // prevent the signature from being too big
-            });
-            this.empty_sign = this.$("#o_portal_signature").jSignature('getData', 'image');
+            this.nameAndSignature = new NameAndSignature(this,
+                options.nameAndSignatureOptions || {});
+
+            // debounce on click to not make multiple RPC if clicking too fast
+            this._onClickSignSubmit = dom.makeButtonHandler(this._onClickSignSubmit);
```

Comment:
Also, if it is async and you do this... it will be a double async which... I don't know what it would do.